src/web3/hooks.ts

```diff
@@ -136,3 +147,42 @@ export const calculateGasMargin = (value: BigNumber): BigNumber => {
   const offset = value.mul(GAS_MARGIN).div(new BigNumber(10000));
   return value.add(offset);
 };
+
+export const useErc20Contract = (
+  address?: string | null,
+): Erc20Detailed | undefined => {
+  const signer = useSigner();
+  return useMemo(
+    () =>
+      signer && address
+        ? Erc20DetailedFactory.connect(address, signer)
+        : undefined,
+    [address, signer],
+  );
+};
+
+export const useSelectedMassetContract = (): Masset | undefined => {
+  const massetState = useSelectedMassetState();
+  const signer = useSigner();
+  return useMemo(
+    () =>
+      signer && massetState?.address
+        ? MassetFactory.connect(massetState.address, signer)
+        : undefined,
+    [massetState, signer],
+  );
+};
+export const useSelectedSaveV1Contract = (): SavingsContract | undefined => {
```

Comment:
It should probably be possible to provide a version as an argument - for next PR